api/server/router/swarm/cluster.go

```diff
@@ -40,5 +40,10 @@ func (sr *swarmRouter) initRoutes() {
 		router.NewPostRoute("/nodes/{id:.*}/update", sr.updateNode),
 		router.NewGetRoute("/tasks", sr.getTasks),
 		router.NewGetRoute("/tasks/{id:.*}", sr.getTask),
+		router.NewGetRoute("/secrets", sr.getSecrets),
+		router.NewPostRoute("/secrets", sr.createSecret),
+		router.NewDeleteRoute("/secrets/{id:.*}", sr.removeSecret),
```

Comment:
Oh, yes, `.` means "any character", but `*` means "zero or more", and `+` means "one or more". Therefore, `.*` would also match `/secrets/` and `/secrets//update` (without id)

The code would probably error out, but we ran into some confusion before (see https://github.com/docker/docker/pull/24634 for reference / more info)